From 44414d45fefd4296f6f06718bcd49acd1033f610 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Mon, 20 May 2024 11:20:19 -0700
Subject: [PATCH 5/6] work in progress to add more gpus

this patch does not yet work, gpu
capabilities needs to be double checked at least.

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 .../triton/python/triton/common/build.py      |  2 +-
 .../triton/python/triton/compiler/compiler.py | 37 +++++++++++++++++++
 v2python/gpu_targets.py                       |  6 +++
 3 files changed, 44 insertions(+), 1 deletion(-)

diff --git a/third_party/triton/python/triton/common/build.py b/third_party/triton/python/triton/common/build.py
index d5551b7..49e7af8 100644
--- a/third_party/triton/python/triton/common/build.py
+++ b/third_party/triton/python/triton/common/build.py
@@ -29,7 +29,7 @@ class CurrentBuildTarget:
     def is_hip():
         aot_arch = CurrentBuildTarget.aot_arch
         if aot_arch is not None:
-            if aot_arch.startswith('MI') or aot_arch.startswith('Radeon'):
+            if aot_arch.startswith('MI') or aot_arch.startswith('Radeon') or aot_arch.startswith('gfx'):
                 return True
             else:
                 return False
diff --git a/third_party/triton/python/triton/compiler/compiler.py b/third_party/triton/python/triton/compiler/compiler.py
index f5b0ff8..8daa81b 100644
--- a/third_party/triton/python/triton/compiler/compiler.py
+++ b/third_party/triton/python/triton/compiler/compiler.py
@@ -388,6 +388,43 @@ def translate_aot_arch(aot_arch):
                  "num_stages": 2,
                  "num_ctas": 1,
                  "warp_size": 64 }, 'hip'
+    if aot_arch == 'gfx1010':
+        mat_core_ver = 3
+        capability = mat_core_ver * 100
+        return { "gfx_triple": 'amdgcn-amd-amdhsa',
+                 "gfx_arch": 'gfx1010',
+                 "gfx_features": '',
+                 "capability": capability,
+                 "matrix_core_version": mat_core_ver,
+                 "num_warps": 4,
+                 "num_stages": 2,
+                 "num_ctas": 1,
+                 "warp_size": 64 }, 'hip'
+    if aot_arch == 'gfx1030':
+        mat_core_ver = 3
+        capability = mat_core_ver * 100
+        return { "gfx_triple": 'amdgcn-amd-amdhsa',
+                 "gfx_arch": 'gfx1030',
+                 "gfx_features": '',
+                 "capability": capability,
+                 "matrix_core_version": mat_core_ver,
+                 "num_warps": 4,
+                 "num_stages": 2,
+                 "num_ctas": 1,
+                 "warp_size": 64 }, 'hip'
+    if aot_arch == 'gfx1035':
+        mat_core_ver = 3
+        capability = mat_core_ver * 100
+        return { "gfx_triple": 'amdgcn-amd-amdhsa',
+                 "gfx_arch": 'gfx1035',
+                 "gfx_features": '',
+                 "capability": capability,
+                 "matrix_core_version": mat_core_ver,
+                 "num_warps": 4,
+                 "num_stages": 2,
+                 "num_ctas": 1,
+                 "warp_size": 64 }, 'hip'
+
     raise NotImplementedError(f'{aot_arch} is not supported as Ahead Of Time target')
 
 def get_cuda_capability(capability):
diff --git a/v2python/gpu_targets.py b/v2python/gpu_targets.py
index 86ef5dc..1c3430b 100644
--- a/v2python/gpu_targets.py
+++ b/v2python/gpu_targets.py
@@ -5,9 +5,15 @@
 AOTRITON_SUPPORTED_GPUS = {
     'MI200'     : 'GPU_ARCH_AMD_GFX90A',
     'MI300X'    : 'GPU_ARCH_AMD_GFX942',
+    'gfx1010'   : 'GPU_ARCH_AMD_GFX1010',
+    'gfx1030'   : 'GPU_ARCH_AMD_GFX1030',
+    'gfx1035'   : 'GPU_ARCH_AMD_GFX1035',
 }
 
 AOTRITON_GPU_ARCH_TUNING_STRING = {
     'MI200'     : 'gfx90a',
     'MI300X'    : 'gfx942',
+    'gfx1010'   : 'GPU_ARCH_AMD_GFX1010',
+    'gfx1030'   : 'GPU_ARCH_AMD_GFX1030',
+    'gfx1035'   : 'GPU_ARCH_AMD_GFX1035',
 }
-- 
2.45.1

